### PR TITLE
chore: do not use Dependabot for npm dependencies

### PR DESCRIPTION
**Pull request by eladb**
We have a custom workflow called “yarn-upgrade” which takes care of continuously updating npm dependencies, so we decided to use Dependabot to only track non-npm dependencies. For this repository, this is basically just GitHub Actions.

Replace the legacy `.dependabot/config.yml` with the native `.github/dependabot.yml` and configure it accordingly.

Also, refactor the auto-approval workflows such that any PR with the `pr/auto-approve` label will get auto-approved by CDK automation. Utilize this new label to auto-approve dependency updates (both from our workflow and the Dependabot ones).

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by RomainMuller — APPROVED** [review]

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Approve PRs with "pr/auto-approve". mergify takes care of the actual merge.
+
+name: auto-approve
+on: pull_request
+
+jobs:
+  auto-approve:
+    if: contains(github.event.pull_request.labels.*.name, 'pr/auto-approve')
```
Comment (nija-at) [inline .github/workflows/auto-approve.yml:8]: Just merging PRs based on a label seems insufficient from a security standpoint. 

Maybe also have a whitelisted set of author accounts (i.e., our known bots)

**Review by nija-at — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-approve.yml (hunk 1 above)
Comment (eladb) [reply .github/workflows/auto-approve.yml:8]: What's the security concern? Applying labels can only be done by administrators.

Comment (nija-at) [reply .github/workflows/auto-approve.yml:8]: Today, administrators cannot approve their own change. This opens that door.

Feels safer to have an allowlist here?

Comment (eladb) [reply .github/workflows/auto-approve.yml:8]: Got it. Okay, I'll add dependabot and cdl-automation in the allow list.

Comment (eladb) [reply .github/workflows/auto-approve.yml]: Okay check out the latest revision please.

File: .github/workflows/auto-approve.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  auto-approve:
+    if: >
+       (
+         contains(github.event.pull_request.labels.*.name, 'pr/auto-approve')
+         && (
+           github.event.pull_request.user.login == 'aws-cdk-automation' 
+           || github.event.pull_request.user.login == 'dependabot[bot]' 
+           || github.event.pull_request.user.login == 'dependabot-preview[bot]'
```
Comment (nija-at) [inline .github/workflows/auto-approve.yml:14]: Would this work or am I expecting too much from GH actions?

```ts
contains(['aws-cdk-automation', 'dependabot[bot]', 'dependabot-preview[bot]'], github.event.pull_request.user.login)
```

**Review by nija-at — APPROVED** [review]

File: .github/workflows/auto-approve.yml (hunk 2 above)
Comment (eladb) [reply .github/workflows/auto-approve.yml:14]: Yes, [that would work](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contains). TIL

**Pull request merged**